pyfftw/interfaces/_utils.py

```diff
@@ -63,21 +63,31 @@ def _Xfftn(a, s, axes, overwrite_input, planner_effort,
 
     if calling_func in ('irfft2', 'irfftn'):
         # overwrite_input is not an argument to irfft2 or irfftn
-        args = (a, s, axes, planner_effort, threads, 
-                auto_align_input, auto_contiguous)
+        args = (planner_effort, threads, auto_align_input, auto_contiguous)
 
         if not overwrite_input:
             # Only irfft2 and irfftn have overwriting the input
             # as the default (and so require the input array to 
             # be reloaded).
-            reload_after_transform = True
+            work_with_copy = True
     else:
-        args = (a, s, axes, overwrite_input, planner_effort, threads, 
+        args = (overwrite_input, planner_effort, threads, 
                 auto_align_input, auto_contiguous)
     
+        if not a.flags.writeable:
+            # Special case of a locked array - always work with a
+            # copy.  See issue #92.
+            work_with_copy = True
+
+    if work_with_copy:
+        # We make the copy before registering the key so that the
+        # copy's stride information will be cached since this will be
+        # used for planning.
+        a = a.copy(order='K')  # Match original layout as closely as possible
```

Comment:
Question: is it better to match the original calling order (C or F) or since we always work with the copy created here, would it be best to make the copy in C order for performance?